2022-09-11 16:49:28,530 : INFO : Start of Training
2022-09-11 16:49:28,530 : INFO : Run_id:2022-09-11_164928_251922471
2022-09-11 16:49:28,530 : INFO : Start of Data Load, validation and transformation
2022-09-11 16:49:28,530 : INFO : Start of Archiving Old Rejected Files...
2022-09-11 16:49:28,530 : INFO : End of Archiving Old Rejected Files...
2022-09-11 16:49:28,530 : INFO : Start of Archiving Old Validated Files...
2022-09-11 16:49:28,531 : INFO : End of Archiving Old Validated Files...
2022-09-11 16:49:28,532 : INFO : Start of Archiving Old Processed Files...
2022-09-11 16:49:28,532 : INFO : End of Archiving Old Processed Files...
2022-09-11 16:49:28,532 : INFO : Start of Archiving Old Result Files...
2022-09-11 16:49:28,532 : INFO : End of Archiving Old Result Files...
2022-09-11 16:49:28,532 : INFO : Start of Reading values From Schema...
2022-09-11 16:49:28,532 : INFO : End of Reading values From Schema...
2022-09-11 16:49:28,532 : INFO : Start of Validating Column Length...
2022-09-11 16:49:28,533 : INFO : End of Validating Column Length...
2022-09-11 16:49:28,533 : INFO : Start of Validating Missing Values...
2022-09-11 16:49:28,533 : INFO : End of Validating Missing Values...
2022-09-11 16:49:28,533 : INFO : Start of Replacing Missing Values with NULL...
2022-09-11 16:49:28,533 : INFO : End of Replacing Missing Values with NULL...
2022-09-11 16:49:28,533 : INFO : Start of Creating Table...
2022-09-11 16:49:28,534 : INFO : Opened training database successfully
2022-09-11 16:49:28,535 : INFO : Tables created successfully
2022-09-11 16:49:28,535 : INFO : Closed training database successfully
2022-09-11 16:49:28,535 : INFO : End of Creating Table...
2022-09-11 16:49:28,536 : INFO : Opened training database successfully
2022-09-11 16:49:28,536 : INFO : Start of Inserting Data into Table...
2022-09-11 16:49:28,536 : INFO : End of Inserting Data into Table...
2022-09-11 16:49:28,536 : INFO : Start of Exporting Data into CSV...
2022-09-11 16:49:28,537 : INFO : Opened training database successfully
2022-09-11 16:49:28,538 : INFO : End of Exporting Data into CSV...
2022-09-11 16:49:28,538 : INFO : Start of Moving Processed Files...
2022-09-11 16:49:28,538 : INFO : End of Moving Processed Files...
2022-09-11 16:49:28,539 : INFO : End of Data Load, validation and transformation
2022-09-11 16:49:28,539 : INFO : Start of Preprocessing...
2022-09-11 16:49:28,539 : INFO : Start of reading dataset...
2022-09-11 16:49:28,545 : INFO : End of reading dataset...
2022-09-11 16:49:28,545 : INFO : Start of Dropping Columns...
2022-09-11 16:49:28,545 : INFO : End of Dropping Columns...
2022-09-11 16:49:28,546 : INFO : Start of feature encoding...
2022-09-11 16:49:28,559 : INFO : End of feature encoding...
2022-09-11 16:49:28,559 : INFO : Start of Dropping Columns...
2022-09-11 16:49:28,559 : INFO : End of Dropping Columns...
2022-09-11 16:49:28,559 : INFO : Start of finding missing values...
2022-09-11 16:49:28,560 : INFO : End of finding missing values...
2022-09-11 16:49:28,560 : INFO : Start of splitting features and label ...
2022-09-11 16:49:28,561 : INFO : End of splitting features and label ...
2022-09-11 16:49:28,561 : INFO : End of Preprocessing...
2022-09-11 16:49:28,562 : INFO : Start of elbow plotting...
2022-09-11 16:49:28,563 : ERROR : Exception raised while elbow plotting:Found array with 0 sample(s) (shape=(0, 5)) while a minimum of 1 is required by KMeans.
Traceback (most recent call last):
  File "C:\Users\napol\PycharmProjects\rappipaychallenge\apps\tuning\cluster.py", line 24, in elbow_plot
    kmeans.fit(data)
  File "C:\Users\napol\PycharmProjects\rappipaychallenge\venv\lib\site-packages\sklearn\cluster\_kmeans.py", line 1367, in fit
    X = self._validate_data(
  File "C:\Users\napol\PycharmProjects\rappipaychallenge\venv\lib\site-packages\sklearn\base.py", line 577, in _validate_data
    X = check_array(X, input_name="X", **check_params)
  File "C:\Users\napol\PycharmProjects\rappipaychallenge\venv\lib\site-packages\sklearn\utils\validation.py", line 909, in check_array
    raise ValueError(
ValueError: Found array with 0 sample(s) (shape=(0, 5)) while a minimum of 1 is required by KMeans.
2022-09-11 16:49:28,566 : ERROR : Unsuccessful End of Training
Traceback (most recent call last):
  File "C:\Users\napol\PycharmProjects\rappipaychallenge\apps\tuning\cluster.py", line 24, in elbow_plot
    kmeans.fit(data)
  File "C:\Users\napol\PycharmProjects\rappipaychallenge\venv\lib\site-packages\sklearn\cluster\_kmeans.py", line 1367, in fit
    X = self._validate_data(
  File "C:\Users\napol\PycharmProjects\rappipaychallenge\venv\lib\site-packages\sklearn\base.py", line 577, in _validate_data
    X = check_array(X, input_name="X", **check_params)
  File "C:\Users\napol\PycharmProjects\rappipaychallenge\venv\lib\site-packages\sklearn\utils\validation.py", line 909, in check_array
    raise ValueError(
ValueError: Found array with 0 sample(s) (shape=(0, 5)) while a minimum of 1 is required by KMeans.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\napol\PycharmProjects\rappipaychallenge\apps\training\train_model.py", line 31, in training_model
    number_of_clusters = self.cluster.elbow_plot(self.X)
  File "C:\Users\napol\PycharmProjects\rappipaychallenge\apps\tuning\cluster.py", line 38, in elbow_plot
    raise Exception()
Exception
